src/stirling/source_connectors/socket_tracer/conn_tracker.h

```diff
@@ -271,9 +271,33 @@ class ConnTracker : NotCopyMoveable {
     CONN_TRACE(2) << absl::Substitute("req_frames=$0 resp_frames=$1", req_frames.size(),
                                       resp_frames.size());
 
-    protocols::RecordsWithErrorCount<TRecordType> result =
-        protocols::StitchFrames<TRecordType, TFrameType, TStateType>(&req_frames, &resp_frames,
-                                                                     state_ptr);
+    protocols::RecordsWithErrorCount<TRecordType> result;
+
+    // If this protocol doesn't support streams, we call StitchFrames with just the deque.
+    // If it does, we use a map of stream ID to deque.
+    if constexpr (TProtocolTraits::stream_support ==
+                  protocols::BaseProtocolTraits<TRecordType>::UseStream) {
+      using TKey = typename TProtocolTraits::key_type;
+      std::map<TKey, std::deque<TFrameType>> requests;
+      std::map<TKey, std::deque<TFrameType>> responses;
+      // TODO(@benkilimnik): Hard code the stream for now. Populate the map in a future PR.
+      requests[0] = std::move(req_frames);
+      responses[0] = std::move(resp_frames);
+      result = protocols::StitchFrames<TRecordType, TKey, TFrameType, TStateType>(
+          &requests, &responses, state_ptr);
+      // TODO(@benkilimnik): Update req and resp frame deques to match maps for now. Populate maps
+      // during parsing in a future PR.
+      req_frames.clear();
+      resp_frames.clear();
```

Comment:
Can you explain why these should be cleared? I believe this will cause us to purge frames that should persist between stitching rounds. I don't know that CQL has a test for that, but I know that was something we saw in the pgsql tests ([background](https://github.com/pixie-io/pixie/issues/697#issuecomment-1411120312)).

Reply:
In lines 284-285 we're transferring ownership of the resources managed by `req_frames` to `requests[0]` and `resp_frames` to `responses[0]`. This leaves `req_frames` in a 'moved-from' state. While it's still a valid object, its state is now unspecified. So, when `protocols::StitchFrames` is called and modifies the deque referenced by `requests[0]`, those changes won't be seen in `req_frames` because `req_frames` has transferred its resources away. I think it is generally not safe to make assumptions about the state of moved-from objects, so I've added these two calls to clear the deques before they are re-populated using the map that was passed in to `StitchFrames`.

Please note that this code is intended to be a stopgap solution until PR 3/3 gets in, which populates the map earlier during parsing.

Reply:
ah, I see. I wasn't sure how the move would work in this case. After looking at the [move operator docs](https://en.cppreference.com/w/cpp/container/deque/operator%3D) and that [.clear()](https://stackoverflow.com/questions/9168823/reusing-a-moved-container) is safe to call on a container in an unspecified state, I understand now.

<img width="1370" alt="Screenshot 2023-10-03 at 1 48 40 PM" src="https://github.com/pixie-io/pixie/assets/5855593/ef924ae4-1a9a-4fc7-bcc2-cb0fa9675bed">